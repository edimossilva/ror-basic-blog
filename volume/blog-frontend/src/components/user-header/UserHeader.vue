<template lang="">
  <div>
    <p>User: {{username}}</p>
    <button v-if="isLoged" button @click="logout">Logout</button>
    <button v-else @click="redirectToLogin">Login</button>
  </div>
</template>

<script>
export default {
  computed: {
    username: function() {
      const username = this.$store.getters.username;
      return username ? username : "not registred";
    },
    isLoged: function() {
      return this.$store.getters.isLoged;
    }
  },
  methods: {
    logout() {
      this.$store.commit("deleteToken");
    },
    redirectToLogin() {
      this.$router.push({ name: "login" });
    }
  }
};
</script>

<style scoped></style>