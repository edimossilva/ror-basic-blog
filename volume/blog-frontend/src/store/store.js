import Vue from 'vue';
import Vuex from 'vuex';

Vue.use(Vuex);

export const store = new Vuex.Store({
  state: {
    authToken: "",
    username: ""
  },
  mutations: {
    saveAuthToken(state, token) {
      state.authToken = token;
    },
    saveUsername(state, username) {
      state.username = username;
    },
    deleteToken(state) {
      state.authToken = "";
    }
  },
  getters: {
    authToken: state => state.authToken,
    username: state => state.username,
    isLoged: state => !!state.authToken
  }
});