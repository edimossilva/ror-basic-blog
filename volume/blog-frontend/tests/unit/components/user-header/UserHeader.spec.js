import { mount, config, createLocalVue } from '@vue/test-utils'

import UserHeader from '@/components/user-header/UserHeader'
import { store } from '@/store/store'

describe('UserHeader', () => {
  it('is a valid component', () => {
    const wrapper = mount(UserHeader, {
      stubs: {
        modal: true,
      },
      store,
    })

    expect(wrapper.isVueInstance()).toBeTruthy()
    expect(wrapper).toMatchSnapshot()
  })
});