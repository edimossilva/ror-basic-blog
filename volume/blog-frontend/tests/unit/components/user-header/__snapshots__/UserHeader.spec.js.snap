// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserHeader is a valid component 1`] = `
<div>
  <p>not registred</p> <button>Login</button>
</div>
`;
